### ISSUE TITLE
sanity checking WHFAST

### ISSUE THREAD

**Issue by dsavransky** [hannorein/rebound#266]
I'm running solar system integrations to sanity check integrator performance, and while the leapfrog gives me the exact stable behavior I'd expect, the same initial conditions and time step cause the WHFAST integration to blow up.  I've been staring at this for a while and have no explanation, so I'd be interested in other opinions.  For various tedious reasons, I'm running this as a mex compiled function in MATLAB, but it's calling the shared library directly so I doubt that that's the issue.  The relevant part of the code:

```
	struct reb_simulation* r = reb_create_simulation();
	// Setup constants
	r->dt = ts[0];
	r->G = 1.0;
	r->ri_whfast.safe_mode 	= 0;		
	r->ri_whfast.corrector 	= 11;		
    if (inttype == 0){
        r->integrator		= REB_INTEGRATOR_LEAPFROG;
    } else{
        r->integrator		= REB_INTEGRATOR_WHFAST;
    }
	r->heartbeat		= heartbeat;
	r->exact_finish_time = 1;
        r->visualization = REB_VISUALIZATION_NONE; 

	// Initial conditions
	for (i=0;i<N;i++){
		struct reb_particle p = {0};
		p.x  = y0[3*i];     p.y  = y0[3*i+1];	p.z  = y0[3*i+2];
		p.vx = dy0[3*i]; 	p.vy = dy0[3*i+1];	p.vz = dy0[3*i+2];
		p.m  = mus[i];
		reb_add(r, p); 
	}
    
	reb_move_to_com(r);
	reb_integrate(r, ts[1]);
```

with heartbeat just writing to an accumulator array to pass back to MATLAB:
```
void heartbeat(struct reb_simulation* r){
	if (reb_output_check(r, ts[2])){
        //reb_exit("asdf");
        //r->t = ts[1]+1.0;
        reb_integrator_synchronize(r);
        t[counter] = (double) r->t;
        struct reb_particle* restrict const particles = r->particles;
        for (i=0;i<N;i++){
            y[counter*N*3 + i*3] = particles[i].x;
            y[counter*N*3 + i*3+1] = particles[i].y;
            y[counter*N*3 + i*3+2] = particles[i].z;
            dy[counter*N*3 + i*3] = particles[i].vx;
            dy[counter*N*3 + i*3+1] = particles[i].vy;
            dy[counter*N*3 + i*3+2] = particles[i].vz;
        }
        counter++;
    }
}
```

Initial conditions are from Horizons and time step is four days.  Outputs as in the figure. Any thoughts? (By the way, if anyone else in the known universe has a use case for REBOUND via MATLAB, I'd be happy to submit a pull).

![whfast_out](https://cloud.githubusercontent.com/assets/4950841/24817725/adb8658e-1bab-11e7-85ad-35fe3dd33e25.png)
![leapfrog_out](https://cloud.githubusercontent.com/assets/4950841/24817726/adb8ba98-1bab-11e7-9f27-c09b8f828199.png)

**Comment by hannorein** [comment]
Hi Dmitry! This is very strange. Your code is almost identical to the outer solar system example (which works fine). Does it make a difference if you turn on safe_mode and turn off symplectic correctors?

**Comment by dsavransky** [comment]
Commenting out the safe_mode and correctors lines has no real effect - the WHFAST integration still blows up.  I compiled and ran the outer system and full solar system packaged examples and they run as expected on my system.  So, I copied out the initial conditions from the solar system example and ran those, and now both the leapfrog and WHFAST generate stable integrations.

So, the problem is clearly with my initial conditions, although I'm still puzzled as to why WHFAST is more sensitive to these than the leapfrog.  Anyway, the integrators appear to be working as they should, so I'll close this.  Thanks.
